lib/Config.cpp

```diff
@@ -63,10 +94,7 @@ void LibConfig::Config::sendJson(const json& data)
   socketLoop->run<uvw::Loop::Mode::ONCE>();
 }
 
-///
-/// \todo get response and set-up name and id
-///
-void LibConfig::Config::loadConfig(KeyWrapper const &givenKey)
+void LibConfig::Config::loadConfig(KeyWrapper givenKey)
```

Comment:
It's unclear to me why taking by copy would be preferable.

Reply:
I dunno why, but the content of KeyWrapper.data was corrupted in the other case, I spent a lot of time searching a solution, but I found nothing. Anyway, this function is called only at the beggining

Reply:
This implies there's probably a bigger problem. I will investigate.

Reply:
Found it, when using a reference there's a potential self-assignment on this line:
https://github.com/raven-os/libconfig/blob/6f3c8d53ae771db6ca088a79182de75e7f3e5ddb/lib/Config.cpp#L105
(when called from here https://github.com/raven-os/libconfig/blob/6f3c8d53ae771db6ca088a79182de75e7f3e5ddb/lib/Config.cpp#L128)
Since the const assignment operator resets the data before copying it here https://github.com/raven-os/libconfig/blob/6f3c8d53ae771db6ca088a79182de75e7f3e5ddb/lib/KeyWrapper.hpp#L52-L53, you end up copying uninitialised memory.
I'd add a check in the assignment operator for self-assignment and take it by const reference here, or I'd do the check before preforming the assignment. Another solution would be not performing the assignment in `LibConfig::Config::loadConfig` itself, but rather require that the assignment be done before and then do the switch on `*key` instead of taking a parameter.